### ISSUE TITLE
Route "/" used `Date.now()` instead of using `performance` ... when using TanStack react query and dynamicIO

### ISSUE THREAD

**Issue by spaceo** [vercel/next.js#73785]
### Link to the code that reproduces this issue

https://github.com/TanStack/query

### To Reproduce

1. Create a query client provider
2. In layout wrap the children in the provider
3. Start the app in dev mode
4. When loading the app the error will occur

Related to this issue:
https://github.com/vercel/next.js/issues/72024

### Current vs. Expected behavior

I would expect no error to happen :)
But apprently the QueryClient uses a "forbidden" way to handle current time in its cache evaluation.

![image](https://github.com/user-attachments/assets/e5f0e3dc-02b5-4104-bc98-404971234e5c)

### Provide environment information

```bash
/bin/sh: pnpm: command not found

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:11 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6020
  Available memory (MB): 16384
  Available CPU cores: 12
Binaries:
  Node: 20.10.0
  npm: 10.2.3
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 15.1.1-canary.0 // Latest available version is detected (15.1.1-canary.0).
  eslint-config-next: 15.0.3
  react: 19.0.0
  react-dom: 19.0.0
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Runtime

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by thatguyinabeanie** [comment]
I am also experiencing this bug as well

**Comment by TechFeatured** [comment]
I was able to remove this error by wrapping my Provider in a server component and calling await connection - then wrapping the call in Suspense.

```
async function QueryProviderOuter({ children }: { children: ReactNode }) {
  await connection();
  return <QueryProvider>{children}</QueryProvider>;
}
```


```
<Suspense fallback={<LoadingSpinner />}>
                        <QueryProviderOuter>
                          <main className="flex-1">{children}</main>
                        </QueryProviderOuter>
                      </Suspense>
```

unsure if this opts the entire subtree into client rendering though

**Comment by gnoff** [comment]
@TechFeatured yeah this will essentially opt everything downstream of this layout into being dynamic only. With PPR you can maybe still have a very generic static shell but since this is presumaly quite high in the application hierarchy no deeper layouts with their own Suspense boundaries will end up being prerendered. I can see that this use of `Date.now()` is related to coped mutations. I don't know yet whether there is an upstream change in react-query that could make this feature compatible with prerendering with dynamicIO. I'll dig in a bit and see if I can make sense of it

**Comment by gnoff** [comment]
I've opened a PR in Tanstack-query that may address this issue if landed. Will revisit once there is movement there

**Comment by Esequiel378** [comment]
Hey! I saw your PR merged a couple of hours ago (excellent work by the way, @gnoff ). Do you have a rough idea of when this would be available? Or if there is a way to use this new change in the meantime 🤗

**Comment by eps1lon** [comment]
Fixed in https://github.com/TanStack/query/pull/8496 which was release in https://github.com/TanStack/query/releases/tag/v5.62.12
